lib/Controller/SettingController.php

```diff
@@ -58,4 +68,17 @@ public function getValue($type) {
 		return $response;
 	}
 
+	protected function setEnableUser (string $userId, $enabled): bool {
+		$success = false;
+		$this->config->setUserValue($this->userId, $this->appName, 'enabled', $enabled);
+		if ($enabled == 'false') {
+			// TODO: Invalidate images and remove user info here???
```

Comment:
@stalker314314 
what you think? Is it a good place to delete user data here?

Reply:
TBH, I would say no. I would split this to 2 separate API calls, one to simply change value, one to do deletion (deletion will be used on some other places, I think (when resetting, when changing model, when admin delete all users data...)).

On the client side, when user uncheck this, I would pop up additional question, like "do you want to keep already detected faces?" and do additional "delete" call if user says no. This way, we get both "pause" and "wipe" semantics.

If you add resetting here in setEnableUser, you are merging them unnecessarly. My thinking, do what you think makes more sense:)

Reply:
> TBH, I would say no. I would split this to 2 separate API calls, one to simply change value, one to do deletion

I think it adds complexity and does not contribute as much.. In this instance, it will remain this way ..
But i want to warn, and ask for confirmation from the user before deleting the data..  But not a second check. If you disable it, you must delete the data ..

> (deletion will be used on some other places, I think (when resetting, when changing model, when admin delete all users data...)).

For this, I will separate the method that performs the cleaning, but it will stay inside the same controller.
If you change the model, it can be done in the same controller, since this controller also can handle admin settings..

> If you add resetting here in setEnableUser, you are merging them unnecessarly. My thinking, do what you think makes more sense:)

To the people that I think with this commit, they do not want to leave the data in the server .. :wink:

Reply:
Ok, it was just a proposal:) I understand your points, makes sense, go with it!;)